senior-knights/course-schedulizer
# Address popover issues

This Branch:
1. Addresses issues in #65 
2. Makes the update popup different from the add popup (gives the option to either update the section, thus deleting the old version, or merely add a new section built off of the section clicked originally)
3. Adds in the missing fields to the popup, except for `startDate`, `endDate`, and `termStart`

To Test:
1. `npm i` and `npm start`
2. Import `math-schedule.csv` and try adding sections the regular way and playing around with updating sections (or adding sections from the update popup)
3. Ensure that updates in things like location, instructor, time, and term are reflected graphically (specifically notice what happens on updating MATH-171-A of Bolt to Spring)
3. Export and verify that changes are reflected

Notes:
1. This should be treated with harsh skepticism in testing because this could unintentionally introduce more issues than before, but in theory, it should make updating much more reliable.
2. This branch more closely associates sections with the instructor, such that MATH-100-A if taught by Bob Smith will be treated separately from MATH-100-A taught by Jane Doe in the same term (this doesn't mean that multiple teachers can't be assigned to a single section though). The reason for this change is that without doing so, changing the term of a class was adding it to another instructor's section of the same class (it's a bit confusing, so ask me if you want more details).
3. This will introduce conflicts with @charkour 's code in #68.
4. Let me know your thoughts on the differentiation of `Update Section` and `Add Section`, is this intuitive or just confusing?

Closes #68
